### DISCUSSION TITLE
USB-UART Terminal port is not working

### DISCUSSION THREAD

**Discussion by maitreyakale** [polarfire-soc/polarfire-soc-documentation#298] [General]
We are trying to achieve Uart interface on the Icicle-Kit but the USB-UART port is not working suddenly,we are unable to open serial terminals both for uart0 and uart1 and when we are giving sudo minicom uart0 and sudo minicom uart1 commands to open the serial terminals,it is showing --> minicom: cannot open /dev/ttyUSB0: No such file or directory and for the uart1 it is the same .

Earlier it was working fine when we are trying to establish ethernet connection!!!
Please suggest any solution for the same.

**Comment by rockoguido** [comment]
hello, I guess is it fair to assume that you are connecting via J11 on the Icicle Kit?   If so then first one needs to make sure that the ttyUSB0 and ttyUSB1 devices are being instantiated.     check your /dev directory, I am thinking they are not being enabled when you plug in the USB cable from the Icicle kit.
2nd you can use the command tail -f /var/log/kern.log to watch what happens when you plug the device in.    The UARTs on the Icicle may be instantiated as ttyACM0/1.   This command will give you insight

I hope that helps.

**Comment by hughbreslin** [comment]
Hi @maitreyakale 

are you still seeing this issue or was it resolved?

**Comment by griffini** [comment]
Hi @maitreyakale

Assuming you have the correct USB cable attached, if you type `ls /dev/ttyUSB*` on your host you should see what devices your system has enumerated. For examplem with the Icicle kit connected to your Linux PC you'll commonly see:

````
$ ls /dev/ttyUS*
/dev/ttyUSB0  /dev/ttyUSB1  /dev/ttyUSB2  /dev/ttyUSB3
````

Sometimes, depending on the state of your USB subsystem (and what devices you had open etc.), and between reboots and unplugging/plugging devices back in, the USB ports on the machine may change or you might seem these enumerate slightly differently on your Linux host.  For example:

````
$ ls /dev/ttyUS*
/dev/ttyUSB2  /dev/ttyUSB3  /dev/ttyUSB4  /dev/ttyUSB5
````

If the Icicle kit is the only USB to virtual UART device connected to your Linux host, you can be pretty certain that the lowest two port numbers correspond to MMUART0 (HSS) and MMUART1 (Linux output) from the Icicle kit.
